frontend/src/components/GeyserStakeView.tsx

```diff
@@ -1,42 +1,77 @@
-import { Wallet } from 'ethers';
-import React, { useContext, useState } from 'react';
-import { MOCK_ERC_20_ADDRESS } from '../constants';
-import { GeyserContext } from '../context/GeyserContext';
-import { VaultContext } from '../context/VaultContext';
-import Web3Context from '../context/Web3Context';
-import { approveCreateDepositStake, approveDepositStake, parseUnitsWithDecimals, permitDepositStake } from '../sdk';
-import { Input } from '../styling/styles';
+import { BigNumber, BigNumberish, Wallet } from 'ethers'
+import { parseUnits } from 'ethers/lib/utils'
+import React, { useContext, useState } from 'react'
+import { GeyserContext } from '../context/GeyserContext'
+import { VaultContext } from '../context/VaultContext'
+import { WalletContext } from '../context/WalletContext'
+import Web3Context from '../context/Web3Context'
+import { approveCreateDepositStake, approveDepositStake } from '../sdk'
+import { NamedColors } from '../styling/colors'
+import { GeyserInteractionButton, Paragraph } from '../styling/styles'
+import { amountOrZero, formatAmount } from '../utils/amount'
+import { PositiveInput } from './PositiveInput'
 
-interface GeyserStakeViewProps {}
+interface Props {}
 
-export const GeyserStakeView: React.FC<GeyserStakeViewProps> = () => {
+export const GeyserStakeView: React.FC<Props> = () => {
   const [amount, setAmount] = useState<string>('')
-  const { selectedGeyser } = useContext(GeyserContext)
+  const [parsedAmount, setParsedAmount] = useState<BigNumber>(BigNumber.from('0'))
+  const { selectedGeyser, stakingTokenDecimals, stakingTokenSymbol } = useContext(GeyserContext)
   const { signer } = useContext(Web3Context)
-  const { selectedVault } = useContext(VaultContext)
+  const { selectedVault, currentLock } = useContext(VaultContext)
+  const { walletAmount, refreshWalletAmount } = useContext(WalletContext)
+
+  const userStake = BigNumber.from(amountOrZero(currentLock?.amount))
+
+  const refresh = () => {
+    setAmount('')
+    setParsedAmount(BigNumber.from('0'))
+    refreshWalletAmount()
+  }
 
   const handleStake = async () => {
-    if (selectedGeyser) {
-      const parsedAmount = await parseUnitsWithDecimals(amount, MOCK_ERC_20_ADDRESS, signer!)
+    if (selectedGeyser && signer && !parsedAmount.isZero()) {
       const geyserAddress = selectedGeyser.id
+      let tx
       if (selectedVault) {
         const vaultAddress = selectedVault.id
-        await approveDepositStake(geyserAddress, vaultAddress, parsedAmount, signer as Wallet)
+        tx = await approveDepositStake(geyserAddress, vaultAddress, parsedAmount, signer as Wallet)
       } else {
-        await approveCreateDepositStake(geyserAddress, parsedAmount, signer as Wallet)
+        tx = await approveCreateDepositStake(geyserAddress, parsedAmount, signer as Wallet)
       }
+      const receipt = await tx.wait()
+      if (receipt) refresh()
```

Comment:
lets store receipt in a state and put the contents of refresh in a useEffect that runs when receipt has changed

Reply:
fixed